File: StarterApp/app/src/main/java/com/elveselimoski/starterapp/di/scope/PerFragment.java
package com.elveselimoski.starterapp.di.scope;

import javax.inject.Scope;
import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

/**
 * Created by Elvedin Selimoski on 12/6/17. mail: elveselimoski@gmail.com
 */
@Scope
@Retention(RetentionPolicy.RUNTIME)
@Target({ElementType.TYPE, ElementType.METHOD})
public @interface PerFragment {
}

File: StarterApp/app/src/main/java/com/elveselimoski/starterapp/di/scope/PerFragment.kt
package com.elveselimoski.starterapp.di.scope

import javax.inject.Scope

/**
 * Created by Elvedin Selimoski on 12/6/17. mail: elveselimoski@gmail.com
 */

@Scope
@Retention
@Target(
    AnnotationTarget.CLASS,
    AnnotationTarget.FILE,
    AnnotationTarget.FUNCTION,
    AnnotationTarget.PROPERTY_GETTER,
    AnnotationTarget.PROPERTY_SETTER
)
annotation class PerFragment